### PR TITLE
Ticket creator

### PR DESCRIPTION
**Pull request by oalfaro2**
Added a feature that will pull target info for a target, as long as it is in the google doc

### CODE REVIEW

File: observation_tickets/Observation_ticket_creator.pyw [hunk 1]
```diff
@@ -52,6 +56,103 @@ def quit_func():
...
+def check_toi():
+    '''
+    Description
+    -----------
+    Checks to see if a target has been selected, and if so,
+    displays it on the widget
+
+    '''
+    global info_directory
+    link = 'https://docs.google.com/spreadsheets/d/17tqqRnaXuI9q_w1uz35wttQQFGpOIP3hhVXZ5-bYnQA/export?format=csv&id=17tqqRnaXuI9q_w1uz35wttQQFGpOIP3hhVXZ5-bYnQA'
```
Comment (Michael-Reefe) [inline observation_tickets/Observation_ticket_creator.pyw:68]: ideally this should be a config parameter defined in config_reader.py, but if you want to keep the ticket creator more as a 'separate' program from the rest of the code, then I suppose I understand.  Still, maybe you could have a pseudo 'ticket creator config' json file in the same observation_ticket directory, that you just read in as a dictionary with json.load or something to have all of these URLs and passwords within

File: observation_tickets/Observation_ticket_creator.pyw [hunk 2]
```diff
@@ -52,6 +56,103 @@ def quit_func():
...
+    '''
+    Description
+    -----------
+    Collects and fills in the target info based on the nights target
+    in the google doc file
+    Returns
+    -------
+
+    '''
+    username = 'tess_nda_observer'
```
Comment (Michael-Reefe) [inline observation_tickets/Observation_ticket_creator.pyw:99]: see above comment about config parameters

File: observation_tickets/Observation_ticket_creator.pyw [hunk 3]
```diff
@@ -52,6 +56,103 @@ def quit_func():
...
+        for row in reader:
+            if row[0] == str(start_date):
+                toi_input = row[2]
+                obs_start = row[6]
+                obs_end = row[7]
+                filter_input = row[8]
+                exposure = row[9]
+    toi = toi_input.split(' ')[1]
+    day = 'today'
+    homeurl_csv = "https://{}:{}@astro.swarthmore.edu/telescope/tess-secure/print_eclipses.cgi?observatory_string=38.828095%3B-77.3056879%3BAmerica%2FNew_York%3BGeorge+Mason+Observatory+%280.8m%29%3BGeorge+Mason&use_utc=0&observatory_latitude=38.828095&observatory_longitude=-77.3056879&timezone=UTC&start_date={}&days_to_print=3&days_in_past=0&minimum_start_elevation=0&and_vs_or=or&minimum_end_elevation=0&minimum_ha=-12&maximum_ha=12&baseline_hrs=0&show_unc=1&maximum_priority=5&minimum_depth=0&maximum_V_mag=&target_string={}&lco_only=0&single_object=0&ra=&dec=&epoch=&period=&duration=&target=&show_ephemeris=0&print_html=2&twilight=-12&max_airmass=2.4".format(
```
Comment (Michael-Reefe) [inline observation_tickets/Observation_ticket_creator.pyw]: config parameter

File: observation_tickets/Observation_ticket_creator.pyw [hunk 4]
```diff
@@ -52,6 +56,103 @@ def quit_func():
...
+    Description
+    -----------
+    Collects and fills in the target info based on the nights target
+    in the google doc file
+    Returns
+    -------
+
+    '''
+    username = 'tess_nda_observer'
+    password = 'F1nd_TE$S_PlaNets!'
```
Comment (Michael-Reefe) [inline observation_tickets/Observation_ticket_creator.pyw]: see above comment about config parameters

File: observation_tickets/Observation_ticket_creator.pyw (hunk 1 above)
Comment (Michael-Reefe) [inline observation_tickets/Observation_ticket_creator.pyw]: ideally this should be a config parameter defined in config_reader.py, but if you want to keep the ticket creator more as a 'separate' program from the rest of the code, then I suppose I understand. Still, maybe you could have a pseudo 'ticket creator config' json file in the same observation_ticket directory, that you just read in as a dictionary with json.load or something to have all of these URLs and passwords within

File: observation_tickets/Observation_ticket_creator.pyw [hunk 5]
```diff
@@ -52,6 +56,103 @@ def quit_func():
...
+    '''
+    global info_directory
+    link = 'https://docs.google.com/spreadsheets/d/17tqqRnaXuI9q_w1uz35wttQQFGpOIP3hhVXZ5-bYnQA/export?format=csv&id=17tqqRnaXuI9q_w1uz35wttQQFGpOIP3hhVXZ5-bYnQA'
+    savefile = requests.get(url=link)
+    current_directory = os.path.abspath(os.path.dirname(__file__))
+    info_directory = os.path.join(current_directory, r'toi_info')
+    try:
+        os.mkdir(info_directory)
+    except:
+        pass
```
Comment (Michael-Reefe) [inline observation_tickets/Observation_ticket_creator.pyw]: I'd replace this with a simple check, the less try/catch statements, the better:
if os.path.exists(info_directory):
  os.makedirs(info_directory)

**Review by Michael-Reefe — CHANGES_REQUESTED** [review]
Looks mostly good!  Just a few small things I wanted to mention.  Otherwise I think it's pretty much ready for whenever @Kakon24 can take a look at it (and all previous PRs).

**Review by Michael-Reefe — COMMENTED** [review]
Looks good!

**Review by Michael-Reefe — APPROVED** [review]

File: observation_tickets/Observation_ticket_creator.pyw [hunk 6]
```diff
@@ -1,24 +1,34 @@
...
+#Loads the urls and passwords needed from url_config.json
+current_directory = os.path.abspath(os.path.dirname(__file__))
+with open(os.path.join(current_directory, 'url_config.json')) as f:
+    url_dict = json.load(f)
+    user = url_dict['Username']
+    passw = url_dict['Password']
 
 def box_labels():
-    """
+    """ss
```
Comment (Kakon24) [inline observation_tickets/Observation_ticket_creator.pyw]: ss needs to be removed

File: observation_tickets/url_config.json [hunk 7]
```diff
@@ -0,0 +1,4 @@
+{"Username": "tess_nda_observer",
+"Password": "F1nd_TE$S_PlaNets",
```
Comment (Kakon24) [inline observation_tickets/url_config.json]: I'm not sure you want this stored in a publicly available repository...

**Pull request merged**